### PR TITLE
Adjust MT-32 settings based on listening tests

### PR DESCRIPTION
**Pull request by kcgen**
This commit will:

1. Move all settings prior to opening the service, as there are _some_ settings that will be ignored until the _next_ open event (and until that happens, they will be ignored).

2. Reduce the per-buffer rendering size from 1024 to 512 frames. This reduces the sysex-to-playback loop from roughly 96 ms to 48 ms, without jeopardizing CPU-limited systems given our asynchronous rendering and keeping a queue of four buffers in backlog. This was tested on a Pi3 with competing load.

3. Use libmtemu32's nice panning behaviour based on listening tests. GranMinigun reported, "It sounds generally more relaxed (easier on ears), and sound scene feels wider, less constrained."

4. Disable libmtemu32's nice partial mixing behaviour based on listening tests. Scanno reported it sounded as if the music has less bass and/or less reverb. The matches the original developer's comment that states the hardware's (non-nice-partial) setting produces a richer sound.

Thanks to:
 - @hw-scanno for flagging this and comparison test builds to help isolate the issue
 - @GranMinigun for provided comparison recordings and critical listening
 -  All authors on this commit for running listening tests and providing feedback.

### CODE REVIEW

**Comment by kcgen** [comment]
Recording and analysis courtesy @GranMinigun:

## Might and Magic III

Intro, city, and dungeon themes. 

"base" recording has nice panning and nice partial mixing disabled. 

https://user-images.githubusercontent.com/1557255/121940505-06f13880-cd03-11eb-8a2c-4c03e78bd78c.mp4

"panning" has nice panning enabled and nice partial mixing disabled. The difference is pretty clear versus "base".

https://user-images.githubusercontent.com/1557255/121940503-06f13880-cd03-11eb-811d-29d4ca2b7161.mp4

"phase" has nice panning disabled and nice partial mixing enabled.

To me, it's too nuanced and situational - city theme, for instance, doesn't sound as good (quiet parts are less vivid), while with intro and dungeon themes I couldn't figure out a clear winner.

https://user-images.githubusercontent.com/1557255/121940499-0658a200-cd03-11eb-9ab7-9bc123bebfec.mp4

## Money Island

Intro theme.  Same recording settings and names as above.

Less drastic difference with nice panning, but still audible. Nice partial mixing seems to make lower frequencies a tad bit more pronounced, but I'm even less certain.

https://user-images.githubusercontent.com/1557255/121940497-05c00b80-cd03-11eb-8ee7-6a2572fbc05e.mp4

https://user-images.githubusercontent.com/1557255/121940496-05277500-cd03-11eb-818f-35717209e995.mp4

https://user-images.githubusercontent.com/1557255/121940490-03f64800-cd03-11eb-9326-0b4d438532fb.mp4

File: src/midi/midi_mt32.cpp [hunk 1]
```diff
@@ -46,25 +46,32 @@
...
 
-// Analog rendering types: BITS16S, FLOAT
+// Analog rendering types: BIT16S, FLOAT
 constexpr auto RENDERING_TYPE = MT32Emu::RendererType_FLOAT;
 
 // Sample rate conversion quality: FASTEST, FAST, GOOD, BEST
 constexpr auto RATE_CONVERSION_QUALITY = MT32Emu::SamplerateConversionQuality_BEST;
 
-// Use improved behavior for volume adjustments, panning, and mixing
+// Prefer higher ramp resolution over the coarser volumer steps used by the hardware
```
Comment (GranMinigun) [inline src/midi/midi_mt32.cpp:67]: Typo: "volumer" vs. "volume".

**Review by GranMinigun — APPROVED** [review]
Looks good.

**Comment by kcgen** [comment]
Thanks for review @GranMinigun !

**Pull request merged**